gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -119,80 +116,95 @@ private static void configureSourceSet(
                         }
                     });
 
-                    // Without this, the 'checkUnusedDependencies correctly picks up project dependency on java-library'
-                    // test fails, by not causing gradle run the jar task, but resolving the path to the jar (rather
-                    // than to the classes directory), which then doesn't exist.
-
-                    // Specifically, we need to pick up the LIBRARY_ELEMENTS_ATTRIBUTE, which is being configured on
-                    // compileClasspath in JavaBasePlugin.defineConfigurationsForSourceSet, but we can't reference it
-                    // directly because that would require us to depend on Gradle 5.6.
-                    // Instead, we just copy the attributes from compileClasspath.
-                    compileClasspath.getAttributes().keySet().forEach(attribute -> {
-                        Object value = compileClasspath.getAttributes().getAttribute(attribute);
-                        conf.getAttributes().attribute((Attribute<Object>) attribute, value);
-                    });
-                });
+                    project.afterEvaluate(_ignored -> {
+                        // Without this, the 'checkUnusedDependencies correctly picks up project dependency on
+                        // java-library'
+                        // test fails, by not causing gradle run the jar task, but resolving the path to the jar (rather
+                        // than to the classes directory), which then doesn't exist.
+                        // Specifically, we need to pick up the LIBRARY_ELEMENTS_ATTRIBUTE, which is being configured on
+                        // compileClasspath in JavaBasePlugin.defineConfigurationsForSourceSet, but we can't reference
+                        // it
+                        // directly because that would require us to depend on Gradle 5.6.
+                        // Instead, we just copy the attributes from compileClasspath.
+                        compileClasspath.get().getAttributes().keySet().forEach(attribute -> {
+                            Object value =
+                                    compileClasspath.get().getAttributes().getAttribute(attribute);
+                            conf.getAttributes().attribute((Attribute<Object>) attribute, value);
+                        });
```

Comment:
There's no way to lazily add a `Map` of attributes (aka lazily conditionally add an attribute). So it too must unfortunately live in an `afterEvaluate`.

Reply:
This was the first thing I tried and it didn't work. Sometimes the library elements attribute does not exist on source configuration. This then causes `attribute(Attribute, Provider<T>)` to throw [as per the javadoc](https://docs.gradle.org/7.6.2/javadoc/org/gradle/api/attributes/AttributeContainer.html#attributeProvider-org.gradle.api.attributes.Attribute-org.gradle.api.provider.Provider-):

> This method can NOT be used to discard the value of an property. Specifying a null provider will result in an IllegalArgumentException being thrown. When the provider has no value at finalization time, an IllegalStateException - regardless of whether or not a convention has been set.